dism resource: add sourcePath parameter to allow for offline feature installs (#1561)
PowerShell/DSC: docs/reference/resources/Microsoft/Windows/OptionalFeatureList/index.md

  - [features](#features) - An array of optional feature entries.

> - **Instance properties:** <a id="instance-properties"></a> The following properties are optional.
>   They define the desired state for an instance of the resource.

Reviewer: ```suggestion
- **Write-only properties:** <a id="write-only-properties"></a> The resource accepts the following
  properties that affect its behavior but can't be retrieved for the actual state of an instance.
  For more information about read-only properties, see the "Write-only resource properties" section
  in [DSC resource properties][aa].
```

| `Possible` | A restart may be required depending on system conditions. |
| `Required` | A restart is required to complete the state change.       |

### sourcePaths

> <details><summary>Expand for <code>sourcePaths</code> property metadata</summary>
> 
> ```yaml
> Type       : array
> IsRequired : false
> IsKey      : false
> IsReadOnly : false
> ```
> 
> </details>
> 
> Supplied at the top level of the **Set** operation, indicates the location of the source files to 
> use for installation if necessary. The DISM API will search these paths if the feature files are
> not available in the local feature store. See the [DISM enable feature documentation][08] for more
> information on valid sources.
> 
> This property is optional and will be omitted from the response if no value is provided.

Reviewer: ``````suggestion
<details><summary>Expand for <code>sourcePaths</code> property metadata</summary>

```yaml
Type              : array
ItemsType         : string
ItemsMinimumCount : 1
ItemsMustBeUnique : true
IsRequired        : false
IsKey             : false
IsReadOnly        : false
IsWriteOnly       : true
```

</details>

Indicates where the resource should look for source files to install a feature when the feature
files aren't available in the local feature store for offline systems. Specify this property as an
array of one or more unique strings where each string represents the path to a source file to use.

This property affects how the resource behaves during **Set** operations. It has no affect for any
other resource operations. For more information about valid sources for the DISM API, see the
[`/Enable-Feature` reference for the `dism` command][08].
``````

        "type": "object",
        "additionalProperties": true
      }
    },    

>     "sourcePaths": {
>       "type": "array",
>       "items": {
>         "type": "string"
>       }
>     },

Reviewer: ```suggestion
    "sourcePaths": {
      "type": "array",
      "items": {
        "type": "string"
      },
      "minItems": 1,
      "uniqueItems": true
    },
```

[04]: ./examples/export-optional-features.md
[05]: ../../../../../concepts/resources/properties.md#read-only-resource-properties
[06]: ../FeatureOnDemandList/index.md
[07]: /windows-hardware/manufacture/desktop/deployment-image-servicing-and-management--dism--command-line-options

> [08]: /windows-hardware/manufacture/desktop/dism-operating-system-package-servicing-command-line-options#enable-feature

Reviewer: ```suggestion
[08]: /windows-hardware/manufacture/desktop/dism-operating-system-package-servicing-command-line-options#enable-feature
[aa]: ../../../../../concepts/resources/properties.md#write-only-resource-properties
```
